projects/step-core/src/lib/modules/basics/components/string-array-input/string-array-input.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, inject, Input, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
+import { ControlValueAccessor, FormBuilder, NgControl } from '@angular/forms';
+import { debounceTime, distinctUntilChanged, map, Subject, takeUntil } from 'rxjs';
+
+type OnChange = (value: string[]) => void;
+type OnTouch = () => void;
+
+@Component({
+  selector: 'step-string-array-input',
+  templateUrl: './string-array-input.component.html',
+  styleUrls: ['./string-array-input.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class StringArrayInputComponent implements OnInit, OnDestroy, ControlValueAccessor {
+  @Input() label?: string;
+  @Input() showRequiredMarker: boolean = false;
+  @Input() tooltip?: string;
+  @Input() errorsDictionary?: Record<string, string>;
+
+  private terminator$ = new Subject<void>();
+  private _fb = inject(FormBuilder).nonNullable;
+
+  private onChange?: OnChange;
+  private onTouch?: OnTouch;
```

Comment:
Why are we using event based architecture instead of observable?

Reply:
@neogucky It's a part of ControlValueAccessor implementation, which is used to make component work with angular's forms (ngModel, formControl, formControlName)